src/fft.rs

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module implements an iterative FFT algorithm for computing the (inverse) Discrete Fourier
+//! Transform (DFT) over a slice of field elements.
+
+use crate::finite_field::FieldElement;
+use crate::util;
+
+/// Sets `outp` to the DFT of `inp`.
+pub fn dft<F: FieldElement>(outp: &mut [F], inp: &[F]) {
+    let n = inp.len();
+    let d = util::log2(n as u128) as usize;
+    let mut u: F;
+    let mut v: F;
+    let mut w: F;
+
+    for i in 0..n {
+        outp[i] = inp[bitrev(d, i)];
+    }
+
+    for l in 1..d + 1 {
+        w = F::root(0).unwrap(); // one
+        let r = F::root(l).unwrap();
+        let y = 1 << (l - 1);
+        for i in 0..y {
+            for j in 0..(n / y) >> 1 {
+                let x = (1 << l) * j + i;
+                u = outp[x];
+                v = w * outp[x + y];
+                outp[x] = u + v;
+                outp[x + y] = u - v;
+            }
+            w *= r;
+        }
+    }
+}
+
+/// Sets `outp` to the inverse of the DFT of `inp`.
+pub fn dft_inv<F: FieldElement>(outp: &mut [F], inp: &[F]) {
```

Comment:
>     1. I'd prefer a longer, more explicit function name like `inverse_discrete_fourier_transform`
Changed to `discrete_fourier_transform_inv`, in order to align with `FieldElement::inv`.

>     2. Would prefer that this return the output as `[F]` instead of taking a mutable slice, unless there's a specific reason to do it this way.
See above comment.